### ISSUE TITLE
Makeflow Containers Cleanup

### ISSUE THREAD

**Issue by dthain** [cooperative-computing-lab/cctools#739]
@charleszheng44 take a second look at the code implementing docker container support in Makeflow.  (I just did an overall cleanup of the code, so note that some names and symbols have changed.

First, there is one serious bug.  `wrapper_input_files` is the list of input files needed by the wrapper for *every* job in the makeflow.

But, I think more importantly, the feature can be much simplified by implementing it differently. Instead of having custom code to write out a new and different wrapper for each node at each job submit, figure out how to write out *one* wrapper (docker.wrapper.sh) when the makeflow starts, and then use the general wrapper feature to accomplish the same thing.

That is, the existing option `--docker-image X` can be implemented by creating `docker.wrapper.sh` and then adding a wrapper command and file, something like this:

```
makeflow_wrapper_add_command("docker.wrapper.sh <image> %%");
makeflow_wrapper_add_input_file("docker.wrapper.sh");
makeflow_wrapper_add_input_file("<image>");
```

This will simplify the code and make docker just a special case of image wrapper.

See what I mean?

Give it a try, issue a pull request, and then I'll take a look.

**Comment by charleszheng44** [comment]
@dthain I noticed that the latest master branch does not include the makeflow sandbox mode. Do you want to keep this feature? Am I supposed to create a new pull request or keep working on the pull request "wq_container_support" ?

**Comment by btovar** [comment]
To make things easier in the future, please create a new pull request based on the current master.

**Comment by dthain** [comment]
To be clear, this issue #739 is regarding makeflow support for docker containers, which is integrated with master, but still needs some improvement.

Your comment above is referring to the makeflow sandbox mode, which is contained in pull #703 which has not been merged due a number of issues that need to be fixed.

**Comment by charleszheng44** [comment]
Since each task may run different command and in multi-core mode, there may be multiple tasks running simultaneously on one machine, just one global script may not work. 

Do you mean that a global docker.wrapper.sh template is created at startup of makeflow, and each task will create its own script based on this template?

**Comment by dthain** [comment]
Makeflow should create one wrapper script at startup.
Every task should use the same wrapper script.
The taskid should be passed as an argument to the wrapper script, so that if it needs to create some auxiliary files, they can be kept unique.

**Comment by dthain** [comment]
Merged as part of #748
